common/src/main/java/org/fao/geonet/utils/BinaryFile.java

```diff
@@ -276,6 +277,19 @@ private String readInput(Path path) {
         }
     }
 
+    private String readInputMinimum(Path path) {
+
+        try {
+            FileInputStream inputStream = new FileInputStream(path.toFile());
+            Scanner sc = new Scanner(inputStream, Constants.CHARSET.toString());
+            String line = sc.nextLine();
+            return line;
+        } catch (IOException e) {
+            Log.error("geonetwork", "Error reading file: " + path);
+            return null;
+        }
+    }
+
```

Comment:
The `FileInputStream` must be closed after reading using the close() method in a `finally` block or declaring the variable in a try-with-resources block (`try(FileInputStream inputStream = new FileInputStream(path.toFile()) { ... } catch() { ...}`

Reply:
Hi @juanluisrp 

I added the close code in the try block. The fileInputStream.close throws IOException itself. So it's better to put it in a try block and catch it in my opinion.

Reply:
Hi @juanluisrp

I added the close code in the try block. The fileInputStream.close throws IOException itself. So it's better to put it in a try block and catch it in my opinion.

Reply:
@juanluisrp 

I added the closure code and it's in a finally block.